File: go/line_test.go
package asu_test

import (
	"testing"

	"github.com/FS-Frost/asu"
	"github.com/stretchr/testify/require"
)

func TestLineContentAssignment(t *testing.T) {
	content := "some-text"
	l := asu.NewLine()
	l.Content = content
	require.Equal(t, content, l.Content)
}

func TestNewLineFromString(t *testing.T) {
	text := `Dialogue: 3,0:00:00.00,1:23:45.67,Default,Chitanda,1,2,3,Some fx,{\b1\fs32\1c&H2F3066&}Some {\1c&H0B0B26&}text`
	expected := asu.NewLine()
	expected.Type = asu.LINE_DIALOGUE
	expected.Layer = 3
	expected.Start = asu.NewTime()
	expected.End = asu.NewTime().FromArgs(1, 23, 45.67)
	expected.Style = "Default"
	expected.Actor = "Chitanda"
	expected.MarginLeft = 1
	expected.MarginRight = 2
	expected.MarginVertical = 3
	expected.Effect = "Some fx"
	expected.Content = `{\b1\fs32\1c&H2F3066&}Some {\1c&H0B0B26&}text`
	actual, err := asu.NewLine().FromString(text)
	require.Nil(t, err)
	require.Equal(t, expected, actual)
}

File: js/tests/line.test.ts
import "../dist/asu";

test("lineContentAssignment", () => {
    const l = asu.newLine();
    const content = "some-text";
    l.content = content;
    expect(l.content).toBe(content);
});

test("newLineFromStringShouldNotFail", () => {
    const text = "Dialogue: 3,0:00:00.00,1:23:45.67,Default,Chitanda,1,2,3,Some fx,{\\b1\\fs32\\1c&H2F3066&}Some {\\1c&H0B0B26&}text";

    let expected = asu.newLine();
    expected.type = "Dialogue";
    expected.layer = 3;
    expected.start = asu.newTime().fromArgs(0, 0, 0);
    expected.end = asu.newTime().fromArgs(1, 23, 45.67);
    expected.style = "Default";
    expected.actor = "Chitanda";
    expected.marginLeft = 1;
    expected.marginRight = 2;
    expected.marginVertical = 3;
    expected.effect = "Some fx";
    expected.content = "{\\b1\\fs32\\1c&H2F3066&}Some {\\1c&H0B0B26&}text";

    const [actual, error] = asu.newLine().fromString(text);
    expect(error).toBeNull();
    expect(actual?.toString()).toStrictEqual(expected.toString());
});

test("newLineFromStringWithInvalidType", () => {
    const text = "Invalid: 3,0:00:00.00,1:23:45.67,Default,Chitanda,1,2,3,Some fx,{\\b1\\fs32\\1c&H2F3066&}Some {\\1c&H0B0B26&}text";

    let expectedError = asu.newError("invalid line type: 'Invalid'");

    const [actualLine, actualError] = asu.newLine().fromString(text);
    expect(actualError).not.toBe(null);
    expect(actualError?.message).toEqual(expectedError.message);
    expect(actualLine).toBeNull();
});
